src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -1454,6 +1454,35 @@ private static IReadOnlyList<DocumentConflict> GetConflictsFor(DocumentsOperatio
             };
         }
 
+        private ChangeVectorEntry[] MergeConflictChangeVectorIfNeededAndDeleteConflicts(ChangeVectorEntry[] documentChangeVector,DocumentsOperationContext context, string key)
+        {
+            ChangeVectorEntry[] mergedChangeVectorEntries = null;
+            bool firstTime = true;
+            int numOfConflicts = 0;
+            foreach (var conflict in GetConflictsFor(context, key))
+            {
+                numOfConflicts++;
+                if (firstTime)
+                {
+                    mergedChangeVectorEntries = conflict.ChangeVector;
+                    firstTime = false;
+                    continue;
+                }
+                mergedChangeVectorEntries = ReplicationUtils.MergeVectors(mergedChangeVectorEntries, conflict.ChangeVector);
+            }
+            //We had conflicts need to delete them
+            if (mergedChangeVectorEntries != null)
+            {
+                DeleteConflictsFor(context, key);
+                Interlocked.Add(ref _hasConflicts,-1*numOfConflicts);
```

Comment:
This might actually cause issues.
The problem is that we immediately make this visible, however, the transaction isn't committed yet.
So it is possible that an read transaction will see the in memory value saying no conflicts, and won't see the conflicts, even though it should (it won't see the resolution).
I suggest doing something like `noConflictsAsOfTxId`, which will be set here if `_hasConflicts` is zeroed.
This way, we'll have to do two checks, the general one, and then one for our trasaction